src/traversals/trees_and_forests.jl

```diff
@@ -0,0 +1,54 @@
+abstract type SpanningTreeAlgorithm end
+
+struct BFS <: SpanningTreeAlgorithm end
+struct RandomBFS <: SpanningTreeAlgorithm end
+struct DFS <: SpanningTreeAlgorithm end
+
+default_spanning_tree_alg() = BFS()
+
+default_root_vertex(g) = last(findmax(eccentricities(g)))
+
+function spanning_tree(g::AbstractNamedGraph; root_vertex=default_root_vertex(g))
+  return spanning_tree(default_spanning_tree_alg(), g; root_vertex)
+end
```

Comment:
```suggestion
function spanning_tree(g::AbstractNamedGraph; alg=default_spanning_tree_alg(), root_vertex=default_root_vertex(g))
  return spanning_tree(alg, g; root_vertex)
end
```
Then users can pass the `alg` as a keyword argument, a bit nicer than having to pass it as the first argument in terms of an external interface.